### ISSUE TITLE
Scrutinizer - Reported Code Injection

### ISSUE THREAD

**Issue by jgrancell** [jgrancell/dash#1]
see https://scrutinizer-ci.com/g/jgrancell/dash/inspections/dc4c7122-7e42-4829-a6b1-0fc729265b08/issues/files/src/classes/Core.class.php?status=new&orderField=path&order=asc&honorSelectedPaths=0#inspectioncomment-38512785

**Comment by jgrancell** [comment]
The payload for 'route' is being flagged as user-input, which is then being used on line 78-79 to call the correct Api class function. Scrutinizer is flagging this as a code injection.

**Tests**

I've attempted injections with the following payloads

POST Request:
`{'apiKey':'pbLf83Y7NXb9130FxekA6jHq','apiVersion':1,'route':'infrastructureGroups(); var_dump('I HACKED YOUR CODE'); die;'}`

GET Request:
`http://localhost/rest/?apiVersion=1&apiKey=pbLf83Y7NXb9130FxekA6jHq&route=infrastructureGroups(); var_dump('I HACKED YOUR CODE'); die;`

**Results**
For both POST and GET, the route payload is being checked by verifyRequest() on line 66, which in turn calls verifyRoute() on line 116. 

verifyRoute() checks the full payload against the routeTable class property. Because this full payload string isn't a route in that routeTable, the following happens: an error is logged by the logger with exceptionCode 201, an error api response is returned to the user with errorCode 201, and the rest of the request fails. The api will never make it past the conditional at line 66.

**Ongoing Testing**
I have added a phpunit test to continue checking this in each build to verify that it returns an errorCode instead of a code injection until we can find a better way to correct this. This test is currently on line 156 of tests/CoreTest.php as the method testCodeInjectionOnRoute()
